### PR TITLE
Improve Popover to work on native

### PR DESCRIPTION
**Pull request by thomasstevens89**
This is my first contribution to Tamagui so I'm looking for feedback before I go any further.

Basically I want to get Popover to work on Native (as an actual popover not a sheet) because it's a useful component on Tablet devices.

I got as far as creating a new `floating` package and refactoring `popper` to use it. I also added a `PopperDemo` just for testing purposes.

If everything is good then I can move on with refactoring popover to use the new `floating` in a more unified way (right now it specifically targets web with `@floating-ui/react`

### CODE REVIEW

**Comment by thomasstevens89** [comment]
Ok Popovers now work on Web and IOS. They are broken when not rendering as a sheet on Android due to the way we use Portals: [https://github.com/tamagui/tamagui/blob/a3deac2c820cf2e263157f27e3c6b82193f549f7/packages/portal/src/Portal.native.tsx#L24](https://github.com/tamagui/tamagui/blob/a3deac2c820cf2e263157f27e3c6b82193f549f7/packages/portal/src/Portal.native.tsx#L24)

**Comment by natew** [comment]
@thomasstevens89 looks incredible on first glance. 

The issue with Android is because you have to manually re-do all context inside the portal. There’s some patterns doing this in Sheet and Dialog for reference.

**Comment by natew** [comment]
Approved the PR so you can get tests and CI passing, and then ping me. Not sure I want to document Popper publicly. at least at the moment

File: packages/popover/src/Popover.tsx [hunk 1]
```diff
@@ -465,7 +486,7 @@ export const Popover = withStaticProperties(
 
     const sheetBreakpoint = when
     const popperScope = usePopoverScope(__scopePopover)
-    const triggerRef = React.useRef<HTMLButtonElement>(null)
```
Comment (natew) [inline packages/popover/src/Popover.tsx:468]: TamaguiElement

File: packages/popper/src/Popper.tsx [hunk 2]
```diff
@@ -105,13 +108,24 @@ export const Popper: React.FC<PopperProps> = (props: ScopedProps<PopperProps>) =
...
+        return
+      }
+      // Only call this when the floating element is rendered
+      return autoUpdate(refs.reference.current, refs.floating.current, floating.update)
+    }, [floating.update, refs.floating, refs.reference])
+  } else {
+    // On Native there's no autoupdate so we call update() when necessary
+    const dimensions = useWindowDimensions();
+    React.useEffect(() => {
+      floating.update();
```
Comment (natew) [inline packages/popper/src/Popper.tsx:123]: have you tested if this also causes the loop or just useMedia?

File: packages/popover/src/Popover.tsx (hunk 1 above)
Comment (thomasstevens89) [reply packages/popover/src/Popover.tsx:468]: Fixed

File: packages/popper/src/Popper.tsx (hunk 2 above)
Comment (thomasstevens89) [reply packages/popper/src/Popper.tsx:123]: It's just useMedia. I turned off the useMedia hook and floating.update() worked without issue. Anyway the good news I handled the issue more upstream and it's no longer crashing.

See [https://github.com/thomasstevens89/tamagui/blob/b34c4d25d176aa58bc414b747ba60733d33caef4/packages/popper/src/Popper.tsx#L315-L318](https://github.com/thomasstevens89/tamagui/blob/b34c4d25d176aa58bc414b747ba60733d33caef4/packages/popper/src/Popper.tsx#L315-L318)

File: packages/popover/src/Popover.tsx [hunk 3]
```diff
@@ -340,44 +344,61 @@ const PopoverContentImpl = React.forwardRef<
...
-                trapped={trapFocus ?? context.open}
-                onMountAutoFocus={onOpenAutoFocus}
-                onUnmountAutoFocus={onCloseAutoFocus}
-              >
-                <div style={{ display: 'contents' }}>{children}</div>
-              </FocusScope>
-            )}
-          </RemoveScroll>
-        </PopperContent>
+        <View key={context.contentId}>
```
[the remark this answers is not in the document]
Comment (thomasstevens89) [reply packages/popover/src/Popover.tsx]: The direct child of `<AnimatePresence>` always needs a key. Anyway I realised we were already using a portal for the content so I moved the dismiss transparent overlay here: [https://github.com/thomasstevens89/tamagui/blob/b34c4d25d176aa58bc414b747ba60733d33caef4/packages/popover/src/Popover.tsx#L261](https://github.com/thomasstevens89/tamagui/blob/b34c4d25d176aa58bc414b747ba60733d33caef4/packages/popover/src/Popover.tsx#L261)

**Comment by thomasstevens89** [comment]
@natew Tests are passing locally.. not sure why the CI failed. All outstanding issues have been resolved, just waiting on your final approval / feedback

File: packages/popper/src/Popper.tsx (hunk 2 above)
Comment (thomasstevens89) [reply packages/popper/src/Popper.tsx:143]: We resolved this, was due to a regression in useMedia that is now fixed

**Comment by natew** [comment]
Ah shit accidental merge.

**Comment by natew** [comment]
Mind re-opening? Sorry about that. Checked out the branch locally and pushing it to a temporary PR here.

Seeing an error on Safari Desktop:


https://user-images.githubusercontent.com/12100/215049541-a97ca018-edf9-4c38-9173-0a0292ab4803.mp4

**Pull request merged**